### DISCUSSION TITLE
Materialize view smart Tuning

### DISCUSSION THREAD

**Discussion by aurelienWls** [ClickHouse/ClickHouse#57185] [Q&A] [answered]
Hi,
I don't find a clear documentation about the following question. I hope that anyone can help me.
In BigQuery when you create a materialize view, if you query the base(s) table(s) of this materialize view, BigQuery automatically rewrites queries to use materialized views whenever possible. See here for more details: https://cloud.google.com/bigquery/docs/materialized-views-use#smart_tuning
I cannot find if this feature exists in Clickhouse. Does anyone know if this is possible in Clickhouse and can point me in the according documentation ?
In my case, I want to migrate an app to clickhouse. Currently, there is a lot of join in each queries perform by my ap. To reduce the rework of the migration, my idea is to not change the actual queries and to create materializes views that is identical to the current queries and hope that clickhouse is smart enough to use the materialize view instead of the bases tables.
Hope i'm clear and already very impressed by all the current feature of Clickhouse.

**Comment by Algunenano** [comment, marked as answer]
ClickHouse can't do that for materialized views. It can do it for [projections](https://clickhouse.com/docs/en/sql-reference/statements/alter/projection) (although not all the time)

**Comment by aurelienWls** [comment]
Thanks for your answer.
I already saw the documentation about [projections](https://clickhouse.com/docs/en/sql-reference/statements/alter/projection). Correct me if i'm wrong, but projection works only on one table. Not when performing join.
I'm going to deep in all possible join optimization and test if it's perform well.

**Reply by Algunenano** [reply]
Yes, projections are only for one table.

Materialized views are triggers that push data into a different table. This target table might contain different data than the query from the MV, for example 2 materialized views dumping data into a single target table, so the optimization that you mention is not possible in ClickHouse.
